libs/liquidity-provision/src/lib/liquidity-provision-provider.tsx

```diff
@@ -0,0 +1,132 @@
+import { useMemo } from 'react';
+import { gql } from '@apollo/client';
+import {
+  makeDataProvider,
+  makeDerivedDataProvider,
+  useDataProvider,
+} from '@vegaprotocol/react-helpers';
+import { Interval } from '@vegaprotocol/types';
+import type { MarketCandles } from '@vegaprotocol/market-list';
+import { marketsCandlesProvider } from '@vegaprotocol/market-list';
+
+import type {
+  LiquidityProvisionMarkets as Markets,
+  LiquidityProvisionMarkets_marketsConnection_edges_node as Market,
+} from './__generated__';
+
+import type { MarketsListData } from './utils';
+import { mapDataToMarketList } from './utils';
+
+export const MARKET_LIST_QUERY = gql`
```

Comment:
chore (blocking): Can you put this query in a .graphql file and utilise the generated hooks

Reply:
Hey @mattrussell36 how do I generate the hooks for this? When I run the command it only creates the types, no hooks. Also, can you point me to any files where the hooks are used? I've seen `. graphql` files in the repo (and the hooks), but I can't find any place where they are actually used.

Reply:
useChainIdQuery in connect-dialog is one example. There aren't many examples of using the hooks directly because this a new utility that is only available as a result of migrating from Apollo's now deprecated built in code generation to @graphql-codegen.

That said, I now see that you are setting up a data provider anyway, so rather than using the hook use the generated query document from the same file, have a look at libs/positions/src/lib/positions-data-providers.ts, its likely you will want something similar.

As to why the hooks aren't being generated, I'm not sure, happy to jump on a call to see whats up

Reply:
This is now working, my bad - I was using Apollo's built in code generation 🤦‍♀️